scripts/verify_mocha_results.js

```diff
@@ -0,0 +1,69 @@
+/* eslint-disable no-console */
+
+// this is a safety script to ensure that Mocha tests ran, by checking:
+// 1. that there are N test results in the reports dir (or at least 1, if N = -1)
+// 2. each of them contains 0 failures and >0 tests
+
+const Bluebird = require('bluebird')
+const fse = Bluebird.promisifyAll(require('fs-extra'))
+const la = require('lazy-ass')
+const path = require('path')
+
+const RESULT_REGEX = /<testsuites name="([^"]+)" time="([^"]+)" tests="([^"]+)" failures="([^"]+)"(?: skipped="([^"]+)"|)>/
```

Comment:
ha, it is simpler to parse XML like this? 
Also, could we output JSON test results instead?

Reply:
yeah, we don't have an xml parser and i figured this would be easier for now

as for outputting json - i assumed that would mess with other systems so i opted to go for this approach